spark-cassandra-connector/src/main/scala/com/datastax/spark/connector/TupleValue.scala

```diff
@@ -0,0 +1,30 @@
+package com.datastax.spark.connector
+
+import scala.reflect.runtime.universe._
+
+import com.datastax.driver.core.{ProtocolVersion, TupleValue => DriverTupleValue}
+import com.datastax.spark.connector.types.NullableTypeConverter
+
+final case class TupleValue(columnValues: IndexedSeq[AnyRef]) extends ScalaGettableByIndexData
+
+object TupleValue {
+
+  def apply(columnValues: Any*): TupleValue =
+    TupleValue(columnValues.map(_.asInstanceOf[AnyRef]).toIndexedSeq)
+
```

Comment:
We don't need unapply but you are right, it would be good to define it to be consistent.